src/components/Dialog/DialogSection.react.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+
+import Dialog from '@material-ui/core/Dialog';
+import uiActions from '../../redux/actions/ui';
+import appActions from '../../redux/actions/app';
+import Share from './Share';
+import Tour from './Tour';
+import SignUp from '../Auth/SignUp/SignUp.react';
+import Login from '../Auth/Login/Login.react';
+import ForgotPassword from '../Auth/ForgotPassword/ForgotPassword.react';
+import ThemeChanger from '../ChatApp/Settings/ThemeChanger';
+import { DialogContainer } from '../Commons/Container';
+
+const DialogData = {
+  share: { component: <Share />, size: 'xs' },
+  login: { component: <Login />, size: 'sm' },
+  signUp: { component: <SignUp />, size: 'sm' },
+  forgotPassword: { component: <ForgotPassword />, size: 'sm' },
+  themeChange: { component: <ThemeChanger />, size: 'md' },
+  tour: { component: <Tour />, size: 'sm' },
+  noComponent: { component: null, size: false },
+};
+
+const DialogSection = props => {
+  const { actions, modalProps, visited } = props;
+
+  const getDialog = () => {
+    if (modalProps && modalProps.isModalOpen) {
+      switch (modalProps.modalType) {
```

Comment:
It is equivalent to `DialogData[modalType]`. No switch required.

Reply:
Neat 💯